### PR TITLE
Feature/disconnect button

### PR DESCRIPTION
**Pull request by juliswer**

### CODE REVIEW

File: src/components/user-dropdown/index.tsx [hunk 1]
```diff
@@ -1,41 +1,54 @@
...
-        </h4>
-      </div>
-      <div className="add-fund-button mt--20 pb--20">
-        <Anchor className="btn btn-primary-alta w-100" path="/connect">
-          Add Your More Funds
-        </Anchor>
+const UserDropdown = () => {
+  const handleDisconnectWallet = async () => {
+    if (typeof window !== undefined) {
+      localStorage.removeItem('authToken');
```
Comment (mmiranda-stackit) [inline src/components/user-dropdown/index.tsx:7]: use the value from the constants instead (constants.tokenKey)

File: src/features/auth/store/auth.actions.ts [hunk 2]
```diff
@@ -43,7 +43,7 @@ const signin = createAsyncThunk<string, string, { rejectValue: void }>(
     const token = await authService.signin(signature, nonce);
 
     localStorageService.setItem(constants.tokenKey, { access_token: token });
-
+    window.location.reload();
```
Comment (mmiranda-stackit) [inline src/features/auth/store/auth.actions.ts:46]: Why are you reloading the page here ?

File: src/layouts/header/index.tsx [hunk 3]
```diff
@@ -20,10 +21,19 @@ type Props = {
...
 const Header = ({ className }: Props) => {
+  const [isSignatured, setIsSignatured] = useState(false);
   const sticky = useSticky();
   const { offcanvas, offcanvasHandler } = useOffcanvas();
   const { search, searchHandler } = useFlyoutSearch();
-  const { connect, connected } = useMetamask();
+  const { connect, connected, sign } = useMetamask();
+
+  useEffect(() => {
+    if (typeof window !== undefined) {
```
Comment (mmiranda-stackit) [inline src/layouts/header/index.tsx:31]: You don't need this. There is a auth slice where you can get the values you need

**Review by mmiranda-stackit — CHANGES_REQUESTED** [review]
See my comments

File: src/layouts/header/index.tsx [hunk 4]
```diff
@@ -20,10 +21,19 @@ type Props = {
...
+      const signature = localStorage.getItem('authToken');
+      if (signature) setIsSignatured(true);
```
Comment (AlefrankM) [inline src/layouts/header/index.tsx:33]: I didn't realize. You can use this like 
`
setIsSignature(!!signature)
`
instead 
`
 if (signature) setIsSignatured(true)....
`

File: src/features/auth/store/auth.actions.ts (hunk 2 above)
Comment (juliswer) [reply src/features/auth/store/auth.actions.ts:46]: If I don't reload the page, the useEffect can't catch up the localStorage change, so the button can't be rendered

File: src/layouts/header/index.tsx (hunk 3 above)
Comment (juliswer) [reply src/layouts/header/index.tsx:31]: I need to get the authToken from the localStorage instead the redux store

File: src/features/auth/store/auth.actions.ts (hunk 2 above)
Comment (mmiranda-stackit) [reply src/features/auth/store/auth.actions.ts:46]: not sure if I get it correctly, but don't use a state variable instead? If you refresh the page, you are getting rid of all the redux state and you are making the ux to behave annoying.

File: src/layouts/header/index.tsx (hunk 3 above)
Comment (mmiranda-stackit) [reply src/layouts/header/index.tsx:32]: Hmm, according to your code, all you need is to know if there is a token in local storage. If the auth slice value 'isAuthenticated' is true, it means that it has valid token which is better yet

**Review by mmiranda-stackit — APPROVED** [review]

**Pull request merged**